libraries/AP_Scripting/drivers/EFI_NMEA2k.lua

```diff
@@ -0,0 +1,196 @@
+--[[
+   EFI driver using NMEA 2000 marine CAN protocol
+--]]
+
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+PARAM_TABLE_KEY = 48
+PARAM_TABLE_PREFIX = "EFI_2K_"
+
+local efi_backend = nil
+local efi_state = EFI_State()
+local cylinder_state = Cylinder_Status()
+if not efi_state or not cylinder_state then
+   return
+end
+
+-- bind a parameter to a variable given
+function bind_param(name)
+    local p = Parameter(name)
+    assert(p, string.format('could not find %s parameter', name))
+    return p
+end
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+    assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+    return bind_param(PARAM_TABLE_PREFIX .. name)
+end
+
+-- Setup EFI Parameters
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 10), 'could not add EFI_2K param table')
```

Comment:
Is there any drawback in declaring a table for 10 positions and using only 3?

Reply:
it consumes a few more bytes, doesn't really matter